account_bank_statement_import_adyen/tests/test_import_adyen.py

```diff
@@ -0,0 +1,58 @@
+from odoo import tests
+from .test_import_file import TestStatementFile
+
+
+@tests.tagged('post_install', 'at_install')
+class TestImportAdyen(TestStatementFile):
+    def setUp(self):
+        super(TestImportAdyen, self).setUp()
+        self.journal = self.env['account.journal'].search(
+            [('type', '=', 'bank')], limit=1)
+        self.journal.default_debit_account_id.reconcile = True
+        self.journal.write({
+            'adyen_merchant_account': 'YOURCOMPANY_ACCOUNT',
+            'update_posted': True,
+            'currency_id': self.ref('base.USD'),
+        })
+
+    def test_import_adyen(self):
+        self._test_statement_import(
+            'account_bank_statement_import_adyen', 'adyen_test.xlsx',
+            'YOURCOMPANY_ACCOUNT 2016/48', self.journal)
+        statement = self.env['account.bank.statement'].search(
+            [], order='create_date desc', limit=1)
+        self.assertEqual(len(statement.line_ids), 22)
+        self.assertTrue(
+            self.env.user.company_id.currency_id.is_zero(
+                sum(line.amount for line in statement.line_ids)))
+
+        account = self.env['account.account'].search([(
+            'internal_type', '=', 'receivable')], limit=1)
+        for line in statement.line_ids:
+            line.process_reconciliation(new_aml_dicts=[{
+                'debit': -line.amount if line.amount < 0 else 0,
+                'credit': line.amount if line.amount > 0 else 0,
+                'account_id': account.id}])
+
+        statement.button_confirm_bank()
+        self.assertEqual(statement.state, 'confirm')
+        lines = self.env['account.move.line'].search([
+            ('account_id', '=', self.journal.default_debit_account_id.id),
+            ('statement_id', '=', statement.id)])
+        reconcile = lines.mapped('full_reconcile_id')
+        self.assertEqual(len(reconcile), 1)
+        self.assertFalse(lines.mapped('matched_debit_ids'))
+        self.assertFalse(lines.mapped('matched_credit_ids'))
```

Comment:
These 2 are now replacement for partial reconciliation ids

Reply:
If I understand correctly, you still get matched debit or credit ids even if there is a full_reconcile_id. I think that is correct: the matched debit/credit ids are not replaced by the full reconcile id. Instead, the full reconcile id is a vehicle to link all of those together, with a currency or other kind of write off line if necessary.

Reply:
Aha. That explains a lot. I will remove the check matched_debit_ids and matched_credit_ids.